### PR TITLE
Tasks changes v2

### PR DESCRIPTION
**Pull request by brunoocasali**
- Add support to filtering and pagination in the tasks resources

### CODE REVIEW

File: spec/meilisearch/client/tasks_spec.rb [hunk 1]
```diff
@@ -24,15 +24,15 @@
     expect(last_task.keys).to include(*succeeded_task_keys)
   end
 
-  it 'gets a task of the MeiliSearch instance' do
+  it 'gets a task of the Meilisearch instance' do
```
Comment (curquiza) [inline spec/meilisearch/client/tasks_spec.rb:27]: Pretty sure we will still find this in 2025 😂

File: lib/meilisearch/task.rb [hunk 2]
```diff
@@ -5,8 +5,13 @@
 
 module MeiliSearch
   class Task < HTTPRequest
-    def task_list
-      http_get '/tasks/'
+    ALLOWED_PARAMS = [:limit, :from, :next, :index_uid, :type, :status].freeze
```
Comment (curquiza) [inline lib/meilisearch/task.rb:8]: `next` is not a parameter, but a field in the response body only

Comment (brunoocasali) [reply lib/meilisearch/task.rb:8]: 🤦 thanks for catching that!

**Review by curquiza — APPROVED** [review]
🪨

**Pull request merged**